server/jetstream_cluster_2_test.go

```diff
@@ -6391,41 +6395,90 @@ func TestJetStreamClusterRePublishUpdateNotSupported(t *testing.T) {
 			if resp.Type != JSApiStreamUpdateResponseType {
 				t.Fatalf("Invalid response type %s expected %s", resp.Type, JSApiStreamUpdateResponseType)
 			}
-			if !IsNatsErr(resp.Error, JSStreamInvalidConfigF) {
-				t.Fatalf("Expected error regarding config error, got %+v", resp.Error)
+			if IsNatsErr(resp.Error, JSStreamInvalidConfigF) {
+				t.Fatalf("Expected no error regarding config error, got %+v", resp.Error)
 			}
 		}
-		expectFailUpdate()
 
-		// Now try with a new stream with RePublish present and then try to change config
-		cfg = &StreamConfig{
-			Name:     stream + "_2",
-			Storage:  MemoryStorage,
-			Replicas: replicas,
-			RePublish: &RePublish{
-				Source:      ">",
-				Destination: "bar.>",
-			},
+		expectRepublished := func(expectedRepub bool) {
+			t.Helper()
+
+			nc, js := jsClientConnect(t, s)
+			defer nc.Close()
+
+			var foos, bars atomic.Int32
+
+			// Create a subscriber for foo.> so that we can see
+			// our published message being echoed back to us.
+			_, err := js.Subscribe(
+				"foo.>",
+				func(msg *nats.Msg) { foos.Add(1) },
```

Comment:
.Add only in go 1.19?

Reply:
Oh yes, I've replaced those with functions that existed in Go 1.18 and also handled the other review comments too.